build(front): landing pages for mentor and mentee added

mzdeloitte opened the pull request:
This is a big one, @ezonai !
- landing pages for `HomeMentee` and `HomeMentor` added
- minor refactors for `RediProgram` and `RediHero`
- `NavBar` edited

Note: 
`RediHeroProgram`:
- `typeSpecificClass` added to avoid redundant code for distinction between `mentor` and `mentee` type
- illustrations on `tablet-only` hidden (decided by Paula)

RediHeroProgram.scss`: 
- `&__content` class written that way to re-use it as well for the `RediHero` component => if bad practise we can outsource it ... idea behind this was to avoid another file for `RediHero` with 4 lines only and that way it works for both components (`RediHero` and `RediHeroProgram`)
- since I use `border-color` for `&--mentee` and `&--mentor` a few times maybe it makes sense to summarize it with an attribute class?
- `&__icon`: height set to have an even line created by the `<hr />` tag ... I am not sure if the entire logic for that line suits you with all the margin, padding etc. but it's the best I could think of :) Happy to re-write if we find a more elegant solution.

`NavBar`:
- compromise between using `NavLink` and add the button classes without using the button. `&__item` class holds styling which is overwritten by `<a>`-tag default styling (tag comes from `NavLink`)
- there is frame of about 60px where the names in the `NavBar` break before the mobile NavBar is visible => if bad practise we will edit
- 

`Button.scss`: 
- `font-weight` deleted (decided by Paula to not have bold text on buttons) => the rest of the changes can be ignored (I guess it's a linter problem again

`SVGImage`
- component holds `SVG`s for `RediProgram` and `RediHeroProgram` ... if suitable we can edit other components which render `SVG`s accordingly

`Carousel`:
- `fade` commented out => Paula prefers slide, I thought fade is nice => decision up to Isabelle

ezonai commented on redi-connect-front/src/components/organisms/RediHeroProgram.tsx:
```diff
@@ -0,0 +1,99 @@
+  type: 'mentor' | 'mentee'
+}
+
+const programIcons = [
```
What about using the `returnObjects` parameter from the i18n? You could move the icon names also to the translation file and make the code much more simple. This way you could completely decouple the content from the component.

https://www.i18next.com/translation-function/objects-and-arrays

ezonai commented on redi-connect-front/src/components/organisms/RediProgram.tsx:
```diff
@@ -21,22 +11,22 @@ const RediProgram = () => {
 
   const programSteps = [
```
This all could be moved with help of the i18n return object to the translation.

mzdeloitte replied:
done. components are much cleaner now

ezonai commented on redi-connect-front/src/components/organisms/RediHeroProgram.tsx:
```diff
@@ -0,0 +1,99 @@
+
+const RediHeroProgram = ({ type }: Props) => {
+  const { t } = useTranslation()
+  const typeSpecificClass = (className: string) => (classnames(`redi-hero__${className}`, {
```
Ok, this you can just replace with one class at the root of the component. You will be then making the component much more easier to read. Since at the end you will be just changing the color which can be managed trough the main class.

ezonai commented on redi-connect-front/src/components/organisms/RediHeroProgram.tsx:
```diff
@@ -0,0 +1,99 @@
+
+  return (
+    <>
+      <Section className={classnames({ [`redi-hero__background--${type}`]: type })}>
```
Block class is missing here. Just name you component first (`redi-user-hero`) and give it a modifier which will decide this type it will have (`redi-user-hero--mentee`). Most of the type specific classes should be removed. 

https://en.bem.info/methodology/

mzdeloitte replied:
edited

mzdeloitte commented on redi-connect-front/src/components/atoms/SVGImage.tsx:
```diff
@@ -1,21 +1,36 @@
 import React from 'react'
-import { ReactComponent as Dragos } from '../../assets/images/profile-dragos.svg'
```
Deleted and implemented in `Icon` component

mzdeloitte commented on redi-connect-front/src/pages/front/landing/HomeMentor.tsx:
```diff
@@ -0,0 +1,24 @@
+  )
+}
+
+export default HomeMentor
```
done

mzdeloitte commented on redi-connect-front/src/routes/routes__logged-out.tsx:
```diff
@@ -35,6 +37,16 @@ export const routes__loggedOut: RouteDefinition[] = [
     exact: true
   },
+  {
+    path: '/front/home/mentor',
```
edited

mzdeloitte commented on redi-connect-front/src/components/organisms/RediHeroProgram.tsx:
```diff
@@ -0,0 +1,99 @@
+  }
+]
+
+const RediHeroProgram = ({ type }: Props) => {
```
edited

mzdeloitte replied:
renamed

mzdeloitte commented on redi-connect-front/src/components/organisms/Carousel.tsx:
```diff
@@ -44,7 +44,7 @@ const Carousel = ({ headline, title, border }: Props) => {
     autoplaySpeed: 10000,
     slidesToShow: 1,
     slidesToScroll: 1,
-    fade: true,
```
Just for testing. Once we decided with which behaviour we want to go we either leave or remove it

mzdeloitte commented:
As suggested I created two main classes: `mentee/mentor-border-color` which I put in the `main.scss` so it can be reused by other components.
Therefore I could also delete the `typeSpecificClass`, instead I defined a `baseClass` for the `border-color` according to the one in the `Button` component

ezonai commented on redi-connect-front/src/components/atoms/Icon.scss:
```diff
@@ -11,7 +11,15 @@
+    height: 4rem;
+  }
+
+  &--default {
```
Does not say a lot about the behaviour it stays for. We also it doesn''t really fit in to sizes... Please check if we could solve this without a generic size like this, since default is not really a size.

mzdeloitte replied:
I was thinking of 3 options: 
1. Another component e.g. `Image` which basically does the same thing than `Icon` but for the "bigger" images like in the hero, landing, prefooter split-menu etc. (also the name would fit better because it's not really an icon anymore)
2. Name it something like `&--original`
3. give the `icon` class `height: 100%` and `width: auto` and remove the default `const iconSize = size || 'medium'`

ezonai commented on redi-connect-front/src/components/organisms/Navbar.tsx:
```diff
@@ -2,16 +2,44 @@ import React, { useState } from 'react'
+const LoggedOutNavItems = () => {
+  const { t } = useTranslation()
+
+  const NavItem = ({ url, children }: NavItemProps) => (
```
Check if we could tune the button to be rendered as Navlink. Would be a nicer solution .

ezonai commented on redi-connect-front/src/components/organisms/RediHero.tsx:
```diff
@@ -18,20 +13,22 @@ const RediHero = () => {
-        <Columns vCentered>
-          <Columns.Column size={5}>
+        <Columns>
+          <Columns.Column size={5} className="redi-hero__content">
```
This says that `redi-hero__content` is a part of `redi-hero`. I don't se the `redi-hero` class anywhere in the the component. 

https://en.bem.info/methodology/

ezonai replied:
I would not recommend separating even class dependencies across multiple components.

mzdeloitte replied:
removed

ezonai commented on redi-connect-front/src/components/organisms/RediHero.tsx:
```diff
@@ -18,20 +13,22 @@ const RediHero = () => {
-        <Columns vCentered>
-          <Columns.Column size={5}>
+        <Columns>
+          <Columns.Column size={5} className="redi-hero__content">
```
Another thing is, please try not to style funcional elements from bulma. This could lead to unpredictable behaviour...

mzdeloitte replied:
removed

mzdeloitte commented:
I added the Image component because those images in the header aren't really icons. If you agree we can re-use the component in the preFooter, rediHero, signupLanding etc.

I added the five-bs class to the main.scss and applied it to the Heading to avoid unpredictable behaviour.

Please also have a look at the Button component ... not sure if that's a better solution but this way we can either render a NavLink or a button.

Last but not least I also renamed the classes in the RediHeroLanding to hero-landing* to be more consistent with the component name.

ezonai commented on redi-connect-front/src/components/atoms/Image.tsx:
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
```
Would be better to call it SVGImage, this way we now exactly what for this component is.

mzdeloitte replied:
renamed

ezonai commented on redi-connect-front/src/components/atoms/Button.tsx:
```diff
@@ -40,6 +47,23 @@ const Button = ({
+}: NavProps) => {
+  return (
+    <NavLink
+      to={to !== undefined ? to : '/'}
```
As far as I see this will never be the case. So for the sake of simplicity remove the condition.

mzdeloitte replied:
`Button.Nav` integrated into `Button`

ezonai commented on redi-connect-front/src/components/organisms/Navbar.tsx:
```diff
@@ -9,9 +9,21 @@ import { useHistory } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
```
Can we change main component name in this document to Navbar as the filename. I know you didn't have anything to do with it but it is still incorrect ATM because ti grew in another direction.

mzdeloitte replied:
`LoggedOutNavbar` renamed to `Navbar`

ezonai commented on redi-connect-front/src/components/organisms/RediHero.tsx:
```diff
@@ -20,18 +15,20 @@ const RediHero = () => {
           <Columns.Column size={5}>
-            <Heading>
-              {t('loggedOutArea.homePage.hero.headline')}
+            <Heading className="five-bs">
```
Please don't define a global class for one time use. Just create a local class for the component.

mzdeloitte replied:
outsourced to `RediHero.scss`

ezonai commented on redi-connect-front/src/components/organisms/RediHeroLanding.tsx:
```diff
@@ -0,0 +1,87 @@
+
+  return (
+    <>
+      <Section className={classnames({ [`hero-landing--${type}`]: type })}>
```
Main class is still a modifier.

mzdeloitte replied:
`section` element added to be consistent with the rest of the css

ezonai commented on redi-connect-front/src/components/organisms/RediHeroLanding.tsx:
```diff
@@ -0,0 +1,87 @@
+import React from 'react'
```
Lets talk about this...
